fsleyes_plugin_shimming_toolbox/utils.py

```diff
@@ -1,24 +1,30 @@
 import subprocess
 import logging
+
+from pathlib import Path
+
 logger = logging.getLogger(__name__)
 
+HOME_DIR = str(Path.home())
+PATH_ST_VENV = f"{HOME_DIR}/shimming-toolbox/python/envs/st_venv/bin"
+
 
-# TODO: find a better way to include this as it is defined in utils as well
 def run_subprocess(cmd):
     """Wrapper for ``subprocess.run()`` that enables to input ``cmd`` as a full string
         (easier for debugging).
 
     Args:
         cmd (string): full command to be run on the command line
     """
-    logging.debug(f'{cmd}')
+    logging.info(f'{cmd}')
     try:
         subprocess.run(
             cmd.split(' '),
             stdout=subprocess.PIPE,
             stderr=subprocess.PIPE,
             text=True,
-            check=True
+            check=True,
+            env=dict(PATH=PATH_ST_VENV)
```

Comment:
~~I think I can offer a cleaner solution:~~ EDIT: Incorrect.

1. The installer for `fsleyes-plugin-shimming-toolbox` uses the install method from the base `shimming-toolbox` project:

    https://github.com/shimming-toolbox/fsleyes-plugin-shimming-toolbox/blob/eadde8445cf4d61e14f9c5648b42b268585c0d01/installer/install_shimming_toolbox.sh#L22-L23

2. And _that_ installer [modifies the shell RC file](https://github.com/shimming-toolbox/shimming-toolbox/blob/f9e272841af5b2798214170b314c342e1174fb79/installer/install_shimming_toolbox.sh#L31) in order to [include the `bin/` directory in the `PATH`](https://github.com/shimming-toolbox/shimming-toolbox/blob/f9e272841af5b2798214170b314c342e1174fb79/installer/install_shimming_toolbox.sh#L21):

    ```bash
    # Update PATH variables based on Shell type
    DISPLAY_UPDATE_PATH="export PATH=\"$ST_DIR/$BIN_DIR:\$PATH\""
   ```

3. Because of this, the existing `PATH` variable _should already contain the `ST_VENV` path!_ So, you just need to pass [`os.environ`](https://docs.python.org/3/library/os.html#os.environ) (the existing set of environment variables with the existing `PATH` variable) instead:

    ```diff
    -           env=dict(PATH=PATH_ST_VENV)
    +           env=os.environ.copy()
    ```

    Otherwise, the default behavior of `subprocess.run` will be to use [`env=None`](https://docs.python.org/3/library/subprocess.html#subprocess.run).

Reply:
Thanks for the looking into this @joshuacwnewton ! I tried your solution (`os.environ.copy()`), but it resulted in the same error as the one obtained in #21... Do you think it is ok to stick to `env=dict(PATH=PATH_ST_VENV)` as it works as expected ?

Reply:
I think it would be alright! But I would like to better understand the root cause, too, since it may reveal other issues with the ST installation method. :thinking:

Reply:
I honestly don't know... The weirdest thing is that the shebang in the `st_b1shim` launcher (and all the other launchers I guess) is only ignored when it links to `st_venv`. When it links to the `base` environment, the execution is indeed performed in `base`, which doesn't make sense to me...

Reply:
> The weirdest thing is that the shebang in the `st_b1shim` launcher (and all the other launchers I guess) is only ignored when it links to `st_venv`

Aha! Thank you for reminding me of this. I think I have an alternate solution:

```suggestion
            shell=True
```

From: https://stackoverflow.com/a/6441590

>

Reply:
That's one of the methods I tried before raising the issue and the shebang was still ignored... Moreover, I saw [here ](https://stackoverflow.com/a/3172690/14061141) that the use of `shell=True`  should be done cautiously. I already spent a lot of time dealing with this problem and the best solution I came up with was to use `env=dict(PATH=PATH_ST_VENV)`, but just like you, I'd like to understand why the other fixes do not work...

Reply:
Ah, my apologies. I wasn't aware of what you have and haven't tried.